racket/collects/net/http-client.rkt

```diff
@@ -63,7 +63,15 @@
   (define ssl-version (if (boolean? ssl?) 'auto ssl?))
 
   (define-values (from to)
-    (cond [ssl?
+    (cond [(list? ssl?)
+           ;; At this point, we have a tunneled socket to the remote host/port: we do not need to
+           ;; address it; ignore host-bs, only use port for conn-port-usual?
+           (match-let (((list ssl? (? input-port? t:from) (? output-port? t:to) abandon-p) ssl?))
```

Comment:
I assume you're saying "change `match-let` to `match-define`" rather than change the whole `cond` into a `match` -- which would also work, but I think would lose some meaning.